Src/IronPython.Modules/_ctypes/CData.cs

```diff
@@ -109,11 +109,23 @@ internal virtual PythonTuple GetBufferInfo()
                 => PythonTuple.MakeTuple(NativeType.TypeFormat, 0, PythonTuple.EMPTY);
 
 
-            #region IBufferProtocol Members
+            #region IBufferProtocol
 
-            IPythonBuffer IBufferProtocol.GetBuffer(BufferFlags flags) => this;
+            private int _numExports; // negative value means object disposed
 
-            void IDisposable.Dispose() { } // TODO
+            IPythonBuffer IBufferProtocol.GetBuffer(BufferFlags flags) {
+                if (_numExports < 0) throw new ObjectDisposedException(GetType().Name);
+                _numExports++;
+                return this;
+            }
+
+            void IDisposable.Dispose() {
+                if (_numExports == 0) {
```

Comment:
> As a side note, I think this violates the description on `IPythonBuffer` which says that "`IDisposable.Dispose()` can be called multiple times (subsequent calls being no-op)"?

This is a very good point; I forgot about multiple-dispose. To satisfy the protocol, I will need to add some mechanism keeping track which export is is disposed when.

> Having a hard time wrapping my head around this. In which scenario does `_numExports` reach 0 and the holder gets disposed?

Given the point above, the mechanism is now moot, but for the record, this is how it was supposed to work:
1. In a trivial scenario, there are no buffer exports, just an ordinary `CData` instance. It is created with `_numExports` equal 0 and when `Dispose()` is called, the holder gets disposed immediately. `_numExports` decrements to -1 and subsequent `GetBuffer` requests throw an ODE.
2. Assume one buffer export is requested, which increases `_numExports` to 1. Then `Dispose` on that export is called, but because `_numExports` is not 0 yet, the holder is not disposed yet either, but `_numExports` drops to 0 and case 1 from above applies further.
3. Assume one buffer export is requested (setting `_numExports` to 1), but before that buffer is released, the whole `CData` object is disposed. This does not dispose the holder yet (because `_numExports` is 1) but sets `_numExports` to 0, making the exported buffer the sole "owner" of the holder. A subsequent call to dispose of the export will dispose the holder (since `_numExports` is 0) and set `_numExport` to -1, marking all data as fully disposed.

This scheme works only on the assumption that the CData instance and each buffer export is disposed only once, which is really dumb because I myself insisted on subsequent `IPythonBuffer.Dispose` calls being idempotent.